modules/margin/src/main/java/com/opengamma/sdk/margin/v3/InvokerMarginClient.java

```diff
@@ -216,6 +221,57 @@ public MarginCalcResult calculate(Ccp ccp, MarginCalcRequest request) {
   }
 
   @Override
+  public MultiCcpMarginCalcResult calculate(List<Ccp> ccps, MarginCalcRequest request) {
+
+    HashMap<Ccp, ScheduledFuture<MarginCalcResult>> futures = new HashMap<>();
+    for (Ccp ccp : ccps) {
+      Callable<MarginCalcResult> ccpAsyncCalculator = () -> calculate(ccp, request);
+      ScheduledFuture<MarginCalcResult> future = invoker.getExecutor().schedule(ccpAsyncCalculator, 5, TimeUnit.SECONDS);
+      futures.put(ccp, future);
+    }
+
+    Map<Ccp, MarginCalcResult> results = new HashMap<>();
+    Map<Ccp, MarginError> completeFailures = new HashMap<>();
+    for (Map.Entry<Ccp, ScheduledFuture<MarginCalcResult>> entry : futures.entrySet()) {
+      try {
+        results.put(entry.getKey(), entry.getValue().get());
+      } catch (InterruptedException | ExecutionException e) {
+        completeFailures.put(entry.getKey(), MarginError.of("TestReason", "Calculator did not return a result for this.", "TestType"));
```

Comment:
"TestReason" ? "TestType" ?

Reply:
Do we have a convention to follow with the reasons and types of errors?